lib/memo_wise.rb

```diff
@@ -179,73 +180,68 @@ def #{method_name}
               end
             end
           END_OF_METHOD
-        else
-          if MemoWise::InternalAPI.has_only_required_args?(method)
-            args_str = method.parameters.map do |type, name|
-              "#{name}#{':' if type == :keyreq}"
-            end.join(", ")
-            args_str = "(#{args_str})"
-            call_str = method.parameters.map do |type, name|
-              type == :req ? name : "#{name}: #{name}"
-            end.join(", ")
-            call_str = "(#{call_str})"
-            fetch_key_params = method.parameters.map(&:last)
-            if fetch_key_params.size > 1
-              fetch_key_init = "[:#{method_name}, #{fetch_key_params.join(', ')}].hash"
-              use_hashed_key = true
-            else
-              fetch_key = fetch_key_params.first.to_s
+        when :one_required_positional, :one_required_keyword
+          # `@_memo_wise_indices` stores the `@_memo_wise_single_argument`
+          # indices of different method names. We only use this data structure
+          # when resetting or presetting memoization. It looks like:
+          #   {
+          #     ClassBeingMemoized => {
+          #       single_arg_method_name: 0,
+          #       other_single_arg_method_name: 1
+          #     },
+          #     ClassBeingMemoizedSingletonClass => {
+          #       another_single_arg_method_name: 0
+          #     }
+          #   }
+          @_memo_wise_indices ||= Hash.new { |h, k| h[k] = {} }
```

Comment:
@ms-ati I'd love your thoughts on this as well.

Reply:
@JacobEvelyn Is your reasoning that this is safe for Marshal, because the Hash with default proc lives in a Class vs in an instance? Do we have tests of the Marshal case that verify this?

Reply:
My reasoning is more one of usage—I find it unlikely that someone would start loading a class (e.g. `memo_wise` one method), then Marshal it, then load it, and then `memo_wise` another method. But maybe I should just change this so we don't worry about it? Adding a test for it feels like it may be overkill though.

Reply:
nvm, I realized this was no longer needed and have removed it.